miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -113,11 +124,30 @@ internal class RealMiniApp(
         }
     }
 
+    private suspend fun verifyManifest(appId: String, versionId: String) {
+        val cachedManifest = downloadedManifestCache.readDownloadedManifest(appId)
+        if (cachedManifest?.versionId != versionId) {
+            val apiManifest = getMiniAppManifest(appId, versionId)
+            downloadedManifestCache.storeDownloadedManifest(appId, CachedManifest(versionId, apiManifest))
+        }
+
+        val customPermissions = miniAppCustomPermissionCache.readPermissions(appId)
+        val manifestPermissions = downloadedManifestCache.getAllPermissions(customPermissions)
+        miniAppCustomPermissionCache.removePermissionsNotMatching(appId, manifestPermissions)
+
+        if (downloadedManifestCache.isRequiredPermissionDenied(customPermissions))
+            throw MiniAppSdkException(ERR_REQUIRED_PERMISSION_DENIED)
```

Comment:
We can create the indicated permission for this exception `MiniAppPermissionException` so Host App can catch it and show the permission dialog again.

Reply:
Yeah, it can be an improvement task.